coalib/results/SourceRange.py

```diff
@@ -158,6 +158,10 @@ def __str__(self):
 
         return format_str.format(self)
 
+    def overlaps(self, other):
+        return (super().overlaps(other) and
+                self.start.file == other.start.file)
```

Comment:
I would put the `self.start.file == other.start.file` before `super().overlaps(other)`, the file-equality check is not so expensive than the base-overlaps check :)